### PR TITLE
sci-libs/dcmtk: version bump and some fixes

### PR DESCRIPTION
**Pull request by fx-carton**

### CODE REVIEW

File: sci-libs/dcmtk/files/dcmtk-3.6.5-rename-bundled-charls.patch [hunk 1]
```diff
@@ -0,0 +1,68 @@
+From 46b4b4c2d48612b60f27969db142599f56042a47 Mon Sep 17 00:00:00 2001
```
Comment (thesamesam) [inline sci-libs/dcmtk/files/dcmtk-3.6.5-rename-bundled-charls.patch:1]: Please link to where the patch is from at the top.

File: sci-libs/dcmtk/dcmtk-3.6.5.ebuild [hunk 2]
```diff
@@ -16,7 +16,6 @@ IUSE="doc png ssl tcpd tiff +threads xml zlib"
 
 RDEPEND="
 	dev-libs/icu:=
-	virtual/jpeg:0
```
[the remark this answers is not in the document]
Comment (fx-carton) [reply sci-libs/dcmtk/dcmtk-3.6.5.ebuild:19]: It uses three versions of IJG's jpeg library : 8, 12, and 16 bits per sample. I remember looking into this for another package in the same situation, and I understand that jpeg-turbo does not support 16 bits at all, and that the current ebuild does not build the 12 bits version. So it doesn't looks like it'd be trivial to use jpeg-turbo instead.

File: sci-libs/dcmtk/dcmtk-3.6.7.ebuild [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+	dev-libs/icu:=
+	png? ( media-libs/libpng:* )
+	ssl? ( dev-libs/openssl:0= )
+	tcpd? ( sys-apps/tcp-wrappers )
+	tiff? ( media-libs/tiff:0 )
+	xml? ( dev-libs/libxml2:2 )
+	zlib? ( sys-libs/zlib )
+"
+DEPEND="${RDEPEND}
+	media-gfx/graphviz
```
[the remark this answers is not in the document]
Comment (fx-carton) [reply sci-libs/dcmtk/dcmtk-3.6.7.ebuild:27]: It's actually not used. I'll remove it.

**Comment by fx-carton** [comment]
@thesamesam I've updated the PR with the requested changes.

Regarding libjpeg, I've looked into it more, and I think I'll unbundle it as a separate package. This will be useful to another package (gdcm) which also depends on the same jpeg-8/12/16 thing (modulo some changes I need to look more in depth to).

File: sci-libs/dcmtk/dcmtk-3.6.7.ebuild [hunk 4]
```diff
@@ -0,0 +1,86 @@
+# Copyright 1999-2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
```
Comment (AndrewAmmerlaan) [inline sci-libs/dcmtk/dcmtk-3.6.7.ebuild:4]: Could we bump this to `EAPI=8` while we are touching the package anyway.

File: sci-libs/dcmtk/dcmtk-3.6.7.ebuild [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+
+inherit cmake
+
+DESCRIPTION="The DICOM Toolkit"
+HOMEPAGE="https://dicom.offis.de/dcmtk.php.en"
+SRC_URI="https://dicom.offis.de/download/dcmtk/release/${P}.tar.gz"
+
+LICENSE="OFFIS"
+KEYWORDS="amd64 ~arm ~arm64 ~ppc64 ~riscv x86"
```
Comment (AndrewAmmerlaan) [inline sci-libs/dcmtk/dcmtk-3.6.7.ebuild:13]: We should drop the stable keywords when adding a new version.

**Comment by AndrewAmmerlaan** [comment]
Great work :+1: Thanks!

**Pull request closed without merging**